server/webapp/WEB-INF/rails.new/webpack/models/preferences/email_settings_model.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+;(function() {
+  "use strict";
+
+  const m = require("mithril");
+  const Stream = require("mithril/stream");
+  const _ = require("lodash");
+
+  function EmailSettingsModel(url, errors) {
+    const email = Stream();
+    const enableNotifications = Stream();
+    const checkinAliases = Stream();
+
+    function trim(string) {
+      return string.replace(/^\s+|\s+$/g, "");
+    }
+
+    function splitter(string) {
+      return _.compact(_.map(string.split(","), trim));
+    }
+
+    const payload = {
+      email:           email,
+      email_me:        enableNotifications,
+      checkin_aliases: checkinAliases.map(splitter)
+    };
+
+    let savedEmail;
+    let savedEnableNotifications;
+    let savedCheckinAliases;
+
+    function fetchUser() {
```

Comment:
See `CrudMixins` and `ElasticProfiles.Profile.get` to see how you can probably push some of this CRUD stuff to helpers.